src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs

```diff
@@ -198,22 +212,41 @@ private static OpenApiSchema CreateCollectionSchema(ODataContext context, OpenAp
                     }
                 }
             };
-            if (context.Settings.EnablePagination)
-            {
-                properties.Add(
-                    "@odata.nextLink",
-                    new OpenApiSchema
-                    {
-                        Type = "string"
-                    });
-            }
 
-            return new OpenApiSchema
+            OpenApiSchema baseSchema = new()
             {
-                Title = $"Collection of {typeName}",
                 Type = "object",
                 Properties = properties
             };
+
+            OpenApiSchema colSchema;
+            if (context.Settings.EnablePagination)
+            {
+                colSchema = new OpenApiSchema
+                {
+                    AllOf = new List<OpenApiSchema>
+                    {
+                        new OpenApiSchema
+                        {
+                            UnresolvedReference = true,
+                            Reference = new OpenApiReference
+                            {
+                                Type = ReferenceType.Schema,
+                                Id = Constants.CollectionPaginationResponse
+                            }
+                        },
+                        baseSchema
+                    }
+                };
+            }
+            else
```

Comment:
I wonder whether the Count property should always be here? Regardless of the pagination setting.
To achieve that we could split the base schema in two BaseCollectionCountResponse and BaseCollectionPagination response. The later is an all of with the former.
Then here depending on the setting, it's either an allOf count response + inline (pagination disabled) or pagination response + inline (pagination enabled).
What do you think?

Reply:
I agree. They are inherently independent, and need to be able to be set and referenced independently.